src/lib/swapper/swappers/ThorchainSwapper/utils/getUsdRate/getUsdRate.ts

```diff
@@ -17,86 +21,104 @@ const USD_POOLS = Object.freeze([
+  return (await thorService.get<ThornodePoolResponse[]>(`${daemonUrl}/lcd/thorchain/pools`))
+    .map(thorPoolsResponse => {
+      const { data } = thorPoolsResponse
+      const availableUsdPools = data.filter(
+        pool => pool.status === 'Available' && USD_POOLS.includes(pool.asset),
+      )
 
-  const availableUsdPools = data.filter(
-    pool => pool.status === 'Available' && USD_POOLS.includes(pool.asset),
-  )
-
-  const { aggregatedRuneUsdPrice, numPools } = availableUsdPools.reduce(
-    (prev, pool) => {
-      const usdBalance = bnOrZero(pool.balance_asset)
-      const runeBalance = bnOrZero(pool.balance_rune)
+      const { aggregatedRuneUsdPrice, numPools } = availableUsdPools.reduce(
+        (prev, pool) => {
+          const usdBalance = bnOrZero(pool.balance_asset)
+          const runeBalance = bnOrZero(pool.balance_rune)
 
-      if (usdBalance.isZero() || runeBalance.isZero()) return prev
+          if (usdBalance.isZero() || runeBalance.isZero()) return prev
 
-      // usdBalance / runeBalance = runeUsdPrice
-      const runeUsdPrice = usdBalance.div(runeBalance)
+          // usdBalance / runeBalance = runeUsdPrice
+          const runeUsdPrice = usdBalance.div(runeBalance)
 
-      prev.aggregatedRuneUsdPrice = prev.aggregatedRuneUsdPrice.plus(runeUsdPrice)
-      prev.numPools++
+          prev.aggregatedRuneUsdPrice = prev.aggregatedRuneUsdPrice.plus(runeUsdPrice)
+          prev.numPools++
 
-      return prev
-    },
-    { aggregatedRuneUsdPrice: bn(0), numPools: 0 },
-  )
+          return prev
+        },
+        { aggregatedRuneUsdPrice: bn(0), numPools: 0 },
+      )
 
-  if (!numPools) {
-    throw new SwapError('[getUsdRate]: no available usd pools', {
-      code: SwapErrorType.USD_RATE_FAILED,
+      return { aggregatedRuneUsdPrice, numPools }
+    })
+    .andThen<BN>(({ aggregatedRuneUsdPrice, numPools }) => {
+      if (!numPools) {
+        return Err(
+          makeSwapErrorRight({
+            message: '[getUsdRate]: no available usd pools',
+            code: SwapErrorType.USD_RATE_FAILED,
+          }),
+        )
+      }
+
+      // aggregatedRuneUsdPrice / numPools = averageRuneUsdPrice
+      return Ok(aggregatedRuneUsdPrice.div(numPools))
     })
-  }
-
-  // aggregatedRuneUsdPrice / numPools = averageRuneUsdPrice
-  return aggregatedRuneUsdPrice.div(numPools)
 }
 
-export const getUsdRate = async (daemonUrl: string, assetId: AssetId): Promise<string> => {
-  try {
-    if (isRune(assetId)) {
-      const runeUsdPrice = await getRuneUsdPrice(daemonUrl)
-      return runeUsdPrice.toFixed(PRECISION)
-    }
+export const getUsdRate = async (
+  daemonUrl: string,
+  assetId: AssetId,
+): Promise<Result<string, SwapErrorRight>> => {
+  if (isRune(assetId)) {
+    const maybeRuneUsdPrice = await getRuneUsdPrice(daemonUrl)
+    return maybeRuneUsdPrice.map(runeUsdPrice => runeUsdPrice.toFixed(PRECISION))
```

Comment:
Missing an `isErr()` before this guy?

Reply:
Nice catch - 0a433773eb0789f0f2886bce3ebd3ca299614281